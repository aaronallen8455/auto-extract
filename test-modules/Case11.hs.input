module Case11 where

import qualified Data.Functor.Identity as Id

test x = let y = Id.Identity 12 in 1 + EXTRACT@simple (x + 3 * Id.runIdentity y)
