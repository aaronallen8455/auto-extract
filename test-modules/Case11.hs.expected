module Case11 where

import qualified Data.Functor.Identity as Id

test x = let y = Id.Identity 12 in 1 + simple x y

simple :: Num a => a -> Id.Identity a -> a
simple x y = x + 3 * Id.runIdentity y
